api/users/views.py

```diff
@@ -252,6 +253,13 @@ class UserDetail(JSONAPIBaseView, generics.RetrieveUpdateAPIView, UserMixin):
 
     serializer_class = UserDetailSerializer
 
+    def get_serializer_class(self):
+        if self.request.auth:
+            scopes = self.request.auth.attributes['accessTokenScope']
+            if ComposedScopes.USER_EMAIL_READ in scopes and self.request.user == self.get_user():
```

Comment:
h/t @icereval for pointing me to `normalize_scopes` which [protects us from future changes](https://github.com/caseyrollins/osf.io/blob/54876e3cb804a3295e64d54abfa1215224bde782/framework/auth/oauth_scopes.py#L272)